### ISSUE TITLE
Xonsh crashes when interrupting running process

### ISSUE THREAD

**Issue by Deviluc** [xonsh/xonsh#2827]
## xonfig

```
+------------------+----------------------+
| xonsh            | 0.7.8                |
| Git SHA          | ecca7a7c             |
| Commit Date      | Sep 11 13:48:32 2018 |
| Python           | 3.7.0                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 2.0.4                |
| shell type       | prompt_toolkit2      |
| pygments         | None                 |
| on posix         | True                 |
| on linux         | True                 |
| distro           | arch                 |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

## Expected Behavior

When running a command that takes longer (or forever, like cat /dev/random) interrupting it (with ctrl-c) will cause the running process to terminate.
## Current Behavior

When running a command that takes longer (or forever, like cat /dev/random) interrupting it (with ctrl-c) will cause the xonsh shell to close.

## Steps to Reproduce

execute ```cat /dev/random```, then ctrl-c

Please note that I'm using xonsh in byoubu (screen) with mutliple xonsh instances open. When the bug occurs, the affected instance (tab) is closed.
I just tested it without screen, the shell window doesn't close but the process (cat) doesn't terminate either.

**Comment by alex-bender** [comment]
yeah, I can't kill process too with `ctrl+c`.

**Comment by alex-bender** [comment]
Will check that tomorrow

On Saturday, September 22, 2018, Anthony Scopatz <notifications@github.com>
wrote:
> Thanks for reporting @Deviluc and @alex-bender. I can't reproduce this
locally. With both prompt-toolkint v2 and readline, Ctrl+C will kill cat
/dev/random and kill the process. Do you have other examples?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.<
https://ci4.googleusercontent.com/proxy/x-CEHUgTi5Q0chmrNB4DetPsVH2y3VU0IH4wM7hcmnODNUw9g1dOqnyXv38eVoOwpA9-gSTk0btfvquD2ll1ZwcLlgW99iUW0ksbgjz-L_Tkc_Tw7CR92NDWFfdSkudJT9Y5Dpk6EMFbQSUbQNAhVgfRnq1VBQ=s0-d-e1-ft#https://github.com/notifications/beacon/ADdfPmF9xIOiBmrqt_0i4Yq1gm25zEruks5udlTsgaJpZM4WwEsO.gif>

**Comment by alex-bender** [comment]
@scopatz indeed, `xonsh` from docker works fine.

**Comment by scopatz** [comment]
Hmm that is weird

**Comment by laloch** [comment]
For me the shell doesn't crash, but becomes unresponsive. Also, `cat /dev/random > test` freezes the shell and the resulting `test` file is empty.

**Comment by scopatz** [comment]
@laloch - that works for me:

```console
scopatz@artemis ~ $ cat /dev/random > test
scopatz@artemis ~ $ cat test
���'�e��Nx\���]�\
```

**Comment by scopatz** [comment]
Any progress on finding another example here?

**Comment by alex-bender** [comment]
@scopatz unfortunately not, nothing new.

**Comment by gsaga** [comment]
This bug is no longer there on master. Should it be closed?

**Comment by laloch** [comment]
> This bug is no longer there on master. Should it be closed?

Can't confirm. `cat /dev/random` still freezes the shell for me with no output whatsoever and no way to SIGINT the subprocess. There's in fact no subprocess at all since `cat` is a builtin. `/bin/cat /dev/random` works just fine.

**Comment by scopatz** [comment]
@laloch - what does `which -a cat` say?

**Comment by laloch** [comment]
It says that it is indeed the xonsh builtin:
```
Type:             function
String form:      <function cat at 0x7f001412b620>
File:             /home/laloch/code/xonsh/xonsh/xoreutils/cat.py
```

**Comment by scopatz** [comment]
Hmm interesting, I wonder if this is an issue with xoreutils cat.

**Comment by scopatz** [comment]
Yeahm it seems that xoreutils cat has this problem.

**Comment by scopatz** [comment]
Potential fix in #2967
